src/main/java/net/minecraftforge/event/entity/player/PlayerEvent.java

```diff
@@ -553,4 +553,46 @@ public void setNewGameMode(GameType newGameMode)
             this.newGameMode = newGameMode;
         }
     }
+
+    /**
+     *
+     * Fired when a player clicks the "done" button in the sign edit UI.
+     * Cancelling will set the edited sign to be blank (i.e. lines will be an empty list)
+     *
+     */
+    @Cancelable
+    public static class SignChangeEvent extends PlayerEvent
+    {
+
+        private final BlockPos pos;
+        private List<String> lines;
+
+        public SignChangeEvent(Player player, BlockPos pos, List<String> lines)
+        {
+            super(player);
+
+            this.pos = pos;
+            this.lines = lines;
+        }
+
+        public BlockPos getPos()
+        {
+            return this.pos;
+        }
+
+        public List<String> getLines()
+        {
+            return this.lines;
```

Comment:
This should be an unmodifiable view of the list since this currently allows event listeners to modify the list directly, bypassing the setter method in the event.